### PR TITLE
Publish indexing metric (total docs) to apm

### PR DESCRIPTION
**Pull request by volodk85**
=== DISCUSS ONLY ===

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/indices/TelemetryAwareIndexingMetrics.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    public static final Setting<TimeValue> TELEMETRY_INDEXING_METRICS_TIME_INTERVAL_SETTING = Setting.positiveTimeSetting(
+        "telemetry.indexing.metrics.time.interval",
+        TimeValue.timeValueSeconds(5),
+        Setting.Property.NodeScope
+    );
+
+    private final ThreadPool threadPool;
+    private final TimeValue interval;
+    private final LongCounter indexedDocsTotalMetric;
+    private final AtomicLong indexedDocsCountAggregator = new AtomicLong(0);
```
Comment (fcofdez) [inline server/src/main/java/org/elasticsearch/indices/TelemetryAwareIndexingMetrics.java:38]: Why are we keeping two counters around? I think it's enough to use `LongCounter` the telemetry module will take care of pushing the metrics asynchronously

File: server/src/main/java/org/elasticsearch/indices/TelemetryAwareIndexingMetrics.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    public void start() {
+        isPublishingEnabled = true;
+        publish();
+    }
+
+    public void stop() {
+        isPublishingEnabled = false;
+    }
+
+    private void publish() {
```
Comment (fcofdez) [inline server/src/main/java/org/elasticsearch/indices/TelemetryAwareIndexingMetrics.java:57]: I think this is not necessary unless I'm missing something. The telemetry library should be lightweight enough to ensure that we don't spend too much time recording the new value.

File: server/src/main/java/org/elasticsearch/indices/TelemetryAwareIndexingMetrics.java (hunk 1 above)
Comment (volodk85) [reply server/src/main/java/org/elasticsearch/indices/TelemetryAwareIndexingMetrics.java:38]: My worries is in super high concurrency which this listener is exposed to. The instance is created in `IndicesService` and passed down to all indices/shards. If we're **absolutely** sure that telemetry's LongCounter can sustain potential load (entire WRITE pool is writing) and *won't become a bottleneck* on a critical path, then we can significantly simply this - no need in collecting atomic long and publishing thread.

File: server/src/main/java/org/elasticsearch/indices/TelemetryAwareIndexingMetrics.java (hunk 2 above)
Comment (volodk85) [reply server/src/main/java/org/elasticsearch/indices/TelemetryAwareIndexingMetrics.java]: True. Verified with core/infra team.

File: server/src/main/java/org/elasticsearch/indices/TelemetryAwareIndexingMetrics.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import org.elasticsearch.index.shard.IndexingOperationListener;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.telemetry.TelemetryProvider;
+import org.elasticsearch.telemetry.metric.LongCounter;
+import org.elasticsearch.telemetry.metric.LongGaugeMetric;
+
+public class TelemetryAwareIndexingMetrics implements IndexingOperationListener {
+
+    private final LongCounter indexingTotalCountMetric;
+    private final LongGaugeMetric indexingDurationTimeMetric;
```
Comment (fcofdez) [inline server/src/main/java/org/elasticsearch/indices/TelemetryAwareIndexingMetrics.java:21]: Is this the right metric type for a duration? that would just publish the latest value every push interval and that's not what we want for this metric.

**Comment by fcofdez** [comment]
@volodk85 If we're just planning to expose the current node stats, we could retrofit them into https://github.com/elastic/elasticsearch/pull/102248

**Pull request closed without merging**